client.bal

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+# Google Drive Client. 
+#
+# + httpClient - The HTTP Client  
+public client class Client {
+    public http:Client httpClient;
+    Configuration driveConfiguration;
+
+    public function init(Configuration driveConfig) returns error? {    
+        self.driveConfiguration = driveConfig;
+        http:ClientSecureSocket? socketConfig = driveConfig?.secureSocketConfig;
+        self.httpClient = check new (BASE_URL, {
+            auth: driveConfig.clientConfig,
+            secureSocket: socketConfig,
+            http1Settings: {chunking: http:CHUNKING_NEVER}
+        });
+    }
+
+    # Gets information about the user, the user's Drive, and system capabilities.
+    # 
+    # + fields - The paths of the fields you want included in the response
+    # + return - If successful, returns `About`. Else returns `error`
+    remote function getAbout(string? fields) returns @tainted About|error {
+        return getDriveInfo(self.httpClient , fields);
+    }
+
+    # Retrieve file using the fileID.
+    # 
+    # + fileId - ID of the file to retreive
+    # + optional - 'GetFileOptional' used to add query parameters to the request
+    # + return - If successful, returns `File`. Else returns `error`
+    remote function getFileById(string fileId, GetFileOptional? optional = ()) returns @tainted File|error {
```

Comment:
Shall we use a better name than optional?

Reply:
Sure. I will change this. This accepts optional query parameters> Any good suggestion for a name?
'optionalParams' ?

Reply:
Changed with meaningful names where possible.. Thanks @LakshanSS